Usetext vitest unit test (#358)
danskernesdigitalebibliotek/dpl-react: docs/code_guidelines.md

  // Another assertion.
  .and("contain", "Material is available and reserved for you!");
```

## Writing Unit Tests

> We are using Vitest as framework for running unit tests.

Reviewer: ```suggestion
We are using [Vitest](https://vitest.dev/) as framework for running unit tests.
```

By using that we can test functions (and therefore also hooks) and classes.

### Where do I place my tests?

> They have to be placed in src/tests/unit.

Reviewer: ```suggestion
They have to be placed in `src/tests/unit`.
```

Reviewer: I wonder if we could place test files along with the code that they test like we do for Cypress tests.

I guess there is a possibility for mixing up the two frameworks if all test files use the format `[scope].test.ts`. If we want to separate the two then one approach might be to use `vite` for `src/core` and Cypress for `src/apps` and `src/components`. That would also clearly communicate the responsibility of the different tools and folders. It might be too strict though.

### Testing hooks

> For testing hooks we are using the library: `@testing-library/react-hooks`
> you can read more about it [here](https://vitest.dev) and you can also take a

Reviewer: ```suggestion
For testing hooks we are using the library [`@testing-library/react-hooks`](https://react-hooks-testing-library.com/) and you can also take a
```

I do not understand how the link to Vitest fits into the current sentence.
